public/pages/CreateMonitor/components/DocumentLevelMonitorQueries/DocumentLevelQuery.js

```diff
@@ -25,11 +33,31 @@ export const ILLEGAL_QUERY_NAME_CHARACTERS = [' '];
 class DocumentLevelQuery extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      fieldDataType: DATA_TYPES.TEXT,
+      indexFieldOptions: [],
+      supportedOperators: getDocLevelQueryOperators(),
+    };
+  }
+
+  componentDidMount() {
+    this.getFieldDataType();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.dataTypes !== this.props.dataTypes) this.getFieldDataType();
+  }
+
+  getFieldDataType() {
```

Comment:
Nit: rename this to `initializeFieldDataType` or `setupFieldDataType`?

Reply:
Sounds good, I'll rename it to `initializeFieldDataType`, and will revise the PR shortly.